doc/user-guide/push-notifications/Push-notifications-client-side.md

```diff
@@ -0,0 +1,158 @@
+          <value>publishers</value>
+        </field>
+      </x>
+    </configure>
+  </pubsub>
+</iq>
+```
+
+The `pubsub.mypubsub.com` will be used as a gateway for all notifications
+and will pass them through to the APNS and/or FCM.
+
+The most important and only difference from the standard node creation
+is the `type='push'` part of the `create` element.
+According to [XEP-0357](https://xmpp.org/extensions/xep-0357.html#enabling), a pubsub node is
+required to route the push notification mechanism. This implies you need a node that will handle
+your push notifications, hence we create a node called `punsub_node_for_my_private_iphone`.
+This node should be unique to the device and you may reuse nodes already created this way.
+It is also important from the security perspective to configure the node with:
+
+* `access_model` set to `whitelist` so only affiliated users can access the node.
+* `publish_model` set to `publishers` so only users with `publisher` or `publisher_only` role
+can publish notifications.
+
+#### Adding the server's JID to allowed publishers
+
+Push notifications to the push node are addressed from your server JID.
+If the push node was configured with the above recommended options, you need to allow your server's
+JID to publish notifications to that node.
+Considering your JID is `alice@mychat.com`, your server's JID is just `mychat.com`.
+The following stanza sent to the just created push node will allow your server JID to publish notifications:
+
+```xml
+<iq to='pubsub.mypubsub.com'
+    type='set'
+    id='wy6Hibg='
+    from='alice@mychat.com/resource'>
+	<pubsub xmlns='http://jabber.org/protocol/pubsub#owner'>
+		<affiliations node='punsub_node_for_my_private_iphone'>
+			<affiliation jid='mychat.com' affiliation='publish-only'/>
+		</affiliations>
+	</pubsub>
+</iq>
+```
+
+### Enabling push notifications
+
+The next and the last step is to enable push notifications on the server that handles your messages
+(and has [mod_event_pusher_push][] enabled).
+Let's assume this server is available under the domain `mychat.com`.
+
+To enable push notifications in the simplest configuration, just send the following stanza:
+
+```xml
+<iq type='set' id='x43'>
+  <enable xmlns='urn:xmpp:push:0' jid='pubsub.mypubsub.com' node='punsub_node_for_my_private_iphone'>
+    <x xmlns='jabber:x:data' type='submit'>
+      <field var='FORM_TYPE'><value>http://jabber.org/protocol/pubsub#publish-options</value></field>
+      <field var='service'><value>apns</value></field>
+      <field var='device_id'><value>your_pns_device_token</value></field>
+      <field var='silent'><value>false</value></field>
+      <field var='topic'><value>some_apns_topic</value></field>
+      <field var='priority'><value>some_priority</value></field>
+    </x>
+  </enable>
+</iq>
+```
+
+We have now enabled push notifications to be send to the `pubsub.mypubsub.com` domain
+on the node `punsub_node_for_my_private_iphone` created previously.
```

Comment:
mention here that node name must be unique for the device. so for the PubSub-less setup the device token received from FCM/APNS can be used to generate (or mb even as is) the unique name

Reply:
@DenysGonchar what do you think about saying that in line 67, where we already mention that the node should be unique to the device?

Reply:
no, it's not good idea. there a statement above:
` If you're running a PubSub-less configuration, skip to this point`
and the line 67 is bypassed!